openscap_report/scap_results_parser/parsers/rule_parser.py

```diff
@@ -146,6 +146,16 @@ def _add_message_about_oval(rule_id, rules):
             msg = "The OVAL graph of the rule as it was displayed before the fix was performed."
             rules[rule_id].messages.append(msg)
 
+    @staticmethod
+    def _get_oval_check_href(rule, rule_result_el):
+        check_ref = rule_result_el.find('.//xccdf:check/xccdf:check-content-ref', NAMESPACES)
+        if check_ref is None:
+            return None
+        check_name = check_ref.get("name")
+        if rule.oval_definition_id is None:
+            rule.oval_definition_id = check_name
```

Comment:
The method is a 'getter', but it *sometimes* mutates the object. On top of that, it is a **static** *private* method. That's very confusing.